scripts/proto-gen.sh

```diff
@@ -0,0 +1,17 @@
+#!/usr/bin/bash
+#
+# Protobuf code generation script.
+#
+# To be run from the root of the Tendermint repository.
+#
+set -euo pipefail
+
+INPUT_DIR=${INPUT_DIR:-./proto}
+PLUGIN=${PLUGIN:-gogofaster}
+PLUGIN_OPTS=${PLUGIN_OPTS:-"Mgoogle/protobuf/timestamp.proto=github.com/gogo/protobuf/types,Mgoogle/protobuf/duration.proto=github.com/golang/protobuf/ptypes/duration,plugins=grpc,paths=source_relative"}
+OUTPUT_DIR=${OUTPUT_DIR:-./proto}
+
+INCLUDES="-I=./third_party/proto/ -I=${INPUT_DIR}"
+PROTOC="protoc ${INCLUDES} --${PLUGIN}_out=${PLUGIN_OPTS}:${OUTPUT_DIR}"
+
+find ${INPUT_DIR} -name '*.proto' -exec ${PROTOC} {} \;
```

Comment:
This seems to do what we need. Not sure if there's a more elegant way?

Reply:
can something simple like https://github.com/cosmos/cosmos-sdk/blob/master/proto/buf.gen.gogo.yaml be down. using a protoc is a pain that is why buf and other tools like protocol came along.